common/utils/filters.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  CatalogueAggregationBucket,
+  CatalogueAggregations,
+} from '@weco/common/model/catalogue';
+import { nodeListValueToArray, inputValue } from '@weco/common/utils/forms';
+import { LinkProps } from 'next/link';
+
+import {
+  worksLink,
+  imagesLink,
+} from '@weco/common/services/catalogue/ts_routes';
+
+export const getFilterByCount = (
+  bucketFilter: CatalogueAggregationBucket[]
+): CatalogueAggregationBucket[] => {
+  return bucketFilter.filter(items => {
+    return items.count > 0;
+  });
+};
+
+export const getAggregationFilterByName = (
+  aggregations: CatalogueAggregations | undefined,
+  name: string
+): CatalogueAggregationBucket[] | [] => {
+  return aggregations && aggregations?.[name]?.buckets
+    ? getFilterByCount(aggregations?.[name]?.buckets)
+    : [];
+};
+
+type RadioGroupOption = {
+  value: string;
+  id: string;
+  label: string;
+};
+
+// sort by the highest count
+export const sortByCount = (
+  a: CatalogueAggregationBucket,
+  b: CatalogueAggregationBucket
+): number => {
+  if (a.count > b.count) {
+    return -1;
+  } else {
+    return 1;
+  }
+};
+
+export const getAggregationRadioGroup = (
+  aggregation: CatalogueAggregationBucket[],
+  prefixId: string
+): RadioGroupOption[] => {
+  const radioOption = aggregation.sort(sortByCount).map(item => {
+    const id = item?.data?.label.replace(/\s/, '-').toLocaleLowerCase();
+    return {
+      value: item?.data?.label,
+      id: `${prefixId}-${id}`,
+      label: `${item?.data?.label} (${item?.count})`,
+    };
+  });
+
+  return radioOption;
+};
+
+export const getFilterItemSelected = (
+  form: HTMLFormElement,
+  formName: string
+): string[] => {
+  const filterCheckboxes = nodeListValueToArray(form[formName]) || [];
+  const selectedFilter = [...filterCheckboxes].filter(
+    selectedFilter => selectedFilter.checked
+  );
+  return selectedFilter.length > 0
+    ? selectedFilter.map(filter => filter.value)
+    : [];
+};
+
+export const getResetFilterLink = (
+  imagesColor: string | null,
+  resetFiltersRoute: any
+): any => {
+  const resetFiltersLink: LinkProps = imagesColor
```

Comment:
This check looks very strange, if there is a `imagesColor`, use the `imageLink`?

Maybe we can uncouple these things a little here and have two methods `resetImageFilterLink` and `resetWorkFiltersLink`?

Can we please type up what resetFilterRoute is here as it seem integral to the method, and I can't fathom it without know what that is.